app/models/binary/DataSet.scala

```diff
@@ -164,8 +185,117 @@ class DataSetDAO @Inject()(sqlClient: SqlClient,
       parsed <- parseFirst(r, id)
     } yield parsed
 
-  def findAllWithSearch(folderIdOpt: Option[ObjectId], searchQuery: Option[String], includeSubfolders: Boolean = false)(
-      implicit ctx: DBAccessContext): Fox[List[DataSet]] =
+  def findAllWithSearch(isActiveOpt: Option[Boolean],
+                        isUnreported: Option[Boolean],
+                        organizationIdOpt: Option[ObjectId],
+                        folderIdOpt: Option[ObjectId],
+                        uploaderIdOpt: Option[ObjectId],
+                        searchQuery: Option[String],
+                        includeSubfolders: Boolean,
+                        limitOpt: Option[Int])(implicit ctx: DBAccessContext): Fox[List[DataSet]] =
+    for {
+      selectionPredicates <- buildSelectionPredicates(isActiveOpt,
+                                                      isUnreported,
+                                                      organizationIdOpt,
+                                                      folderIdOpt,
+                                                      uploaderIdOpt,
+                                                      searchQuery,
+                                                      includeSubfolders)
+      limitQuery = limitOpt.map(l => q"LIMIT $l").getOrElse(q"")
+      r <- run(q"SELECT $columns FROM $existingCollectionName WHERE $selectionPredicates $limitQuery".as[DatasetsRow])
+      parsed <- parseAll(r)
+    } yield parsed
+
+  def findAllCompactWithSearch(isActiveOpt: Option[Boolean],
+                               isUnreported: Option[Boolean],
+                               organizationIdOpt: Option[ObjectId],
+                               folderIdOpt: Option[ObjectId],
+                               uploaderIdOpt: Option[ObjectId],
+                               searchQuery: Option[String],
+                               requestingUserIdOpt: Option[ObjectId],
+                               includeSubfolders: Boolean,
+                               limitOpt: Option[Int])(implicit ctx: DBAccessContext): Fox[List[DatasetCompactInfo]] =
+    for {
+      selectionPredicates <- buildSelectionPredicates(isActiveOpt,
+                                                      isUnreported,
+                                                      organizationIdOpt,
+                                                      folderIdOpt,
+                                                      uploaderIdOpt,
+                                                      searchQuery,
+                                                      includeSubfolders)
+      limitQuery = limitOpt.map(l => q"LIMIT $l").getOrElse(q"")
+      query = q"""
+            SELECT
+              d._id,
+              d.name,
+              o.name,
+              d._folder,
+              d.isUsable,
+              d.displayName,
+              d.created,
+              COALESCE(
```

Comment:
Btw this could also be implemented as a view: a view of all dataset+user combinations with an attribute whether the user can read/write the dataset. The benefit of a view would be that we can easily examine it with pgadmin et al and it could potentially be used by multiple queries.

Reply:
I agree about the views. However, we need to figure out a way to handle them during migrations. The existing userInfos view is already frequently a source of trouble (needs to be dropped + recreated in evolutions). I would not like to have to do this for two more views all the time :grimacing:  Maybe there is a way to get around this.

Reply:
What's the problem with dropping/recreating?

Reply:
Not a conceptual problem, but time and again we wrote evolutions with only their semantics in mind, then saw them fail because the view handling was not yet included. It of course makes the evolutions a lot bigger as well. None of this totally blocks this, but I would only go for it if we see concrete benefit from creating more views